### PR TITLE
Added ReSharper MustUseReturnValue attribute to A.CallTo()

### LINKED ISSUE #633: add MustUseReturnValue to `A.CallTo` and `NextCall.To`
Proposed by @drauch in https://github.com/FakeItEasy/FakeItEasy/issues/490#issuecomment-199825874.

We hadn't cared for the idea of putting `Pure` on the methods back when discussing #505, but `MustUseReturnValue` has exactly the semantics we want.

@thomaslevesque has determined that this would present a duplicate warning to one that the Analyzer discussed in #490 would, but there's no real harm.

### PR DESCRIPTION
**Pull request by drauch**
Fixes #633.

### CODE REVIEW

File: src/FakeItEasy/A.cs [hunk 1]
```diff
@@ -80,6 +81,7 @@ public static T Dummy<T>()
         /// </summary>
         /// <param name="callSpecification">An expression where the configured member is called.</param>
         /// <returns>A configuration object.</returns>
+        [MustUseReturnValue("Must specify an assertion as well, e.g., .MustHaveHappened()")]
```
Comment (ulrichb) [inline src/FakeItEasy/A.cs]: Justification text: Maybe "CallTo() makes no sense without following _setup or_ assertion."?

Comment (drauch) [inline src/FakeItEasy/A.cs]: Improved messages (force pushed changes -> see new commits).

**Comment by thomaslevesque** [comment]
Thanks for the PR @drauch.
Nitpicking: rather than repeating the same message every time in the attribute, could you introduce a constant? (I don't think it's possible to use resources for this)
Otherwise, everything looks good to me

**Comment by blairconrad** [comment]
If we're going to provide this service for `A.CallTo`, I think we should for `NextCall.To` as well.

**Comment by drauch** [comment]
@thomaslevesque : I've extracted a constant (own class, so that it can be reused within `NextCall.To`).

@blairconrad : Done.

File: src/FakeItEasy/JetBrainsAnnotationsJustificationMessages.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+namespace FakeItEasy
+{
+    using JetBrains.Annotations;
+
+    /// <summary>
+    /// Common justification messages for JetBrains ReSharper annotations.
+    /// </summary>
+    public static class JetBrainsAnnotationsJustificationMessages
```
Comment (blairconrad) [inline src/FakeItEasy/JetBrainsAnnotationsJustificationMessages.cs]: Should probably be `internal`, like the attribute. There's no need to expose this.

**Comment by blairconrad** [comment]
@drauch, thanks for the adjustment. I've made one more comment about the visibility of `JetBrainsAnnotationsJustificationMessages`. How do you feel about making the change and squashing the commits? There's been a little bit of back and forth what with our comments and the standalone commit for the 3.5 build and whatnoat.

**Comment by adamralph** [comment]
I've been away on vacation so chiming in later here, but I'm still :-1: on us shipping R# annotations. I'll add my reasoning over at #490

**Comment by drauch** [comment]
@blairconrad : Sure can do, however, not before you've come to a final verdict whether to merge this branch or not. Please, keep me posted.

**Comment by adamralph** [comment]
@FakeItEasy/owners I propose that we close this, citing the [Resharper Gallery](https://github.com/FakeItEasy/FakeItEasy/issues/490#issuecomment-202840961) as the solution for R# annotations.

**Comment by blairconrad** [comment]
@adamralph, based on https://github.com/FakeItEasy/FakeItEasy/issues/490#issuecomment-199825874, I don't think there's anything else to do.

**Comment by adamralph** [comment]
@drauch I'm closing this as a `wontfix` since @FakeItEasy/owners have come to the consensus that we prefer that the community to use the [Resharper Gallery](https://resharper-plugins.jetbrains.com/packages) as the solution for Resharper annotations (more at https://github.com/FakeItEasy/FakeItEasy/issues/490#issuecomment-202840961).

We are grateful for your contribution and we appreciate the time you put into it. It is always with some reluctance that we close a community PR without merging it. However, having commits merged into master is by no means the only method of contribution. Your proposal has driven out a valuable conversation which is a significant contribution in itself.

**Comment by blairconrad** [comment]
As @adamralph said, @drauch, thanks very much for your work, and I'm additionally sorry that it took us a while to reach the decision we did.

I would still be happy to collaborate on an external annotations pack, if you're interested.

**Comment by blairconrad** [comment]
Hey, @drauch! I've some promising results from an external annotations pack.

image

**Pull request closed without merging**